### ISSUE TITLE
AttributeError at /admin/login/ - 'LoginView' object has no attribute 'get_redirect_url'

### ISSUE THREAD

**Issue by markfinger** [wagtail/wagtail#4838]
Hi, a bug was recently introduced that introduces a hard crash when visiting the admin login view as an anon user. 

Related commit: https://github.com/wagtail/wagtail/commit/aac33358834ce77e221b0b26db1af36898a7e9c8

I don't know the larger context, but I'd guess that adding a `hasattr` check should suffice as a workaround/fix.

We can replicate this with wagtail 2.2.2 and 2.3. We've downgraded to 2.2.1 to avoid it.

Error
```AttributeError at /admin/login/
'LoginView' object has no attribute 'get_redirect_url'
```

Stack
```....
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/utils/decorators.py", line 67, in _wrapper
    return bound_func(*args, **kwargs)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/views/decorators/debug.py", line 76, in sensitive_post_parameters_wrapper
    return view(request, *args, **kwargs)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/utils/decorators.py", line 63, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/utils/decorators.py", line 67, in _wrapper
    return bound_func(*args, **kwargs)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/utils/decorators.py", line 149, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/utils/decorators.py", line 63, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/utils/decorators.py", line 67, in _wrapper
    return bound_func(*args, **kwargs)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/views/decorators/cache.py", line 57, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/utils/decorators.py", line 63, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/contrib/auth/views.py", line 90, in dispatch
    return super(LoginView, self).dispatch(request, *args, **kwargs)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/wagtail/admin/views/account.py", line 213, in get
    return super().get(*args, **kwargs)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/views/generic/edit.py", line 174, in get
    return self.render_to_response(self.get_context_data())
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/wagtail/admin/views/account.py", line 219, in get_context_data
    context = super().get_context_data(**kwargs)
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/django/contrib/auth/views.py", line 124, in get_context_data
    self.redirect_field_name: self.get_success_url(),
  File "/Users/markfinger/Projects/agsa/var/go.sh-venv/lib/python3.6/site-packages/wagtail/admin/views/account.py", line 206, in get_success_url
    return self.get_redirect_url() or reverse('wagtailadmin_home')
```

**Comment by chosak** [comment]
@markfinger thanks for the bug report. Can you confirm the version of Django that you're using?

It looks to me like LoginView.get_redirect_url wasn't added until [this change](https://github.com/django/django/commit/16431b03f801788d791bbb24d6fb266c3591ab07) in Django 1.11.3 (to fix [this ticket](https://code.djangoproject.com/ticket/28229)), also released to the master (Django 2.0+) branch [here](https://github.com/django/django/commit/e7dc39fb65e51d7613c941f7e5768b621dea4e76).

If you're on Django 1.11 < 1.11.3, could you try upgrading to a newer patch version of 1.11 to see if that fixes this for you with the latest releases of Wagtail?

**Comment by markfinger** [comment]
Thanks, @chosak. I've upgraded from `django==1.11` to `django==1.11.16`, and the problems are no longer evident with `wagtail==2.2.2`.
